llvm/test/CodeGen/RISCV/rvv/fixed-vectors-masked-load-fp.ll

```diff
@@ -450,29 +2106,2892 @@ define void @masked_load_v64f32(ptr %a, ptr %m_ptr, ptr %res_ptr) nounwind {
+; RV64-ZVFHMIN-NEXT:    vmv.x.s a1, v10
+; RV64-ZVFHMIN-NEXT:    fmv.h.x fa4, a1
+; RV64-ZVFHMIN-NEXT:    fcvt.s.h fa4, fa4
+; RV64-ZVFHMIN-NEXT:    feq.s a1, fa4, fa5
+; RV64-ZVFHMIN-NEXT:    vslidedown.vi v10, v8, 14
+; RV64-ZVFHMIN-NEXT:    vmv.x.s a4, v10
+; RV64-ZVFHMIN-NEXT:    fmv.h.x fa4, a4
+; RV64-ZVFHMIN-NEXT:    fcvt.s.h fa4, fa4
+; RV64-ZVFHMIN-NEXT:    feq.s a4, fa4, fa5
+; RV64-ZVFHMIN-NEXT:    vslidedown.vi v10, v8, 13
+; RV64-ZVFHMIN-NEXT:    vmv.x.s a5, v10
+; RV64-ZVFHMIN-NEXT:    fmv.h.x fa4, a5
+; RV64-ZVFHMIN-NEXT:    fcvt.s.h fa4, fa4
+; RV64-ZVFHMIN-NEXT:    feq.s a5, fa4, fa5
+; RV64-ZVFHMIN-NEXT:    vslidedown.vi v10, v8, 12
+; RV64-ZVFHMIN-NEXT:    vmv.x.s a6, v10
+; RV64-ZVFHMIN-NEXT:    fmv.h.x fa4, a6
+; RV64-ZVFHMIN-NEXT:    fcvt.s.h fa4, fa4
+; RV64-ZVFHMIN-NEXT:    feq.s a6, fa4, fa5
+; RV64-ZVFHMIN-NEXT:    sb a6, 12(sp)
+; RV64-ZVFHMIN-NEXT:    sb a5, 13(sp)
+; RV64-ZVFHMIN-NEXT:    sb a4, 14(sp)
+; RV64-ZVFHMIN-NEXT:    sb a1, 15(sp)
+; RV64-ZVFHMIN-NEXT:    vslidedown.vi v10, v8, 11
+; RV64-ZVFHMIN-NEXT:    vmv.x.s a1, v10
+; RV64-ZVFHMIN-NEXT:    fmv.h.x fa4, a1
+; RV64-ZVFHMIN-NEXT:    fcvt.s.h fa4, fa4
+; RV64-ZVFHMIN-NEXT:    feq.s a1, fa4, fa5
+; RV64-ZVFHMIN-NEXT:    vslidedown.vi v10, v8, 10
+; RV64-ZVFHMIN-NEXT:    vmv.x.s a4, v10
+; RV64-ZVFHMIN-NEXT:    fmv.h.x fa4, a4
+; RV64-ZVFHMIN-NEXT:    fcvt.s.h fa4, fa4
+; RV64-ZVFHMIN-NEXT:    feq.s a4, fa4, fa5
+; RV64-ZVFHMIN-NEXT:    vslidedown.vi v10, v8, 9
+; RV64-ZVFHMIN-NEXT:    vmv.x.s a5, v10
+; RV64-ZVFHMIN-NEXT:    fmv.h.x fa4, a5
+; RV64-ZVFHMIN-NEXT:    fcvt.s.h fa4, fa4
+; RV64-ZVFHMIN-NEXT:    feq.s a5, fa4, fa5
+; RV64-ZVFHMIN-NEXT:    vslidedown.vi v8, v8, 8
+; RV64-ZVFHMIN-NEXT:    vmv.x.s a6, v8
+; RV64-ZVFHMIN-NEXT:    fmv.h.x fa4, a6
+; RV64-ZVFHMIN-NEXT:    fcvt.s.h fa4, fa4
+; RV64-ZVFHMIN-NEXT:    feq.s a6, fa4, fa5
+; RV64-ZVFHMIN-NEXT:    sb a6, 8(sp)
+; RV64-ZVFHMIN-NEXT:    sb a5, 9(sp)
+; RV64-ZVFHMIN-NEXT:    sb a4, 10(sp)
+; RV64-ZVFHMIN-NEXT:    sb a1, 11(sp)
+; RV64-ZVFHMIN-NEXT:    addi a1, sp, 64
+; RV64-ZVFHMIN-NEXT:    vsetvli zero, a3, e8, m4, ta, ma
+; RV64-ZVFHMIN-NEXT:    vle8.v v8, (a1)
+; RV64-ZVFHMIN-NEXT:    mv a1, sp
+; RV64-ZVFHMIN-NEXT:    vle8.v v12, (a1)
+; RV64-ZVFHMIN-NEXT:    vand.vi v16, v8, 1
+; RV64-ZVFHMIN-NEXT:    vmsne.vi v8, v16, 0
+; RV64-ZVFHMIN-NEXT:    vand.vi v12, v12, 1
+; RV64-ZVFHMIN-NEXT:    vmsne.vi v0, v12, 0
+; RV64-ZVFHMIN-NEXT:    addi a1, a0, 128
+; RV64-ZVFHMIN-NEXT:    vle16.v v16, (a1), v0.t
+; RV64-ZVFHMIN-NEXT:    vmv1r.v v0, v8
+; RV64-ZVFHMIN-NEXT:    vle16.v v8, (a0), v0.t
+; RV64-ZVFHMIN-NEXT:    vse16.v v8, (a2)
+; RV64-ZVFHMIN-NEXT:    addi a0, a2, 128
+; RV64-ZVFHMIN-NEXT:    vse16.v v16, (a0)
+; RV64-ZVFHMIN-NEXT:    addi sp, s0, -512
+; RV64-ZVFHMIN-NEXT:    ld ra, 504(sp) # 8-byte Folded Reload
+; RV64-ZVFHMIN-NEXT:    ld s0, 496(sp) # 8-byte Folded Reload
+; RV64-ZVFHMIN-NEXT:    addi sp, sp, 512
+; RV64-ZVFHMIN-NEXT:    ret
   %m = load <128 x half>, ptr %m_ptr
```

Comment:
This is more of a comment on the test structure than anything else, and definitely non-blocking.

Most of the run lines in these files appear to come from the mask generation, not the actual masked load or store.  Maybe you should just load a mask from memory or pass one in as a param instead?  Testing the mask generation is covered (presumably) in other test files, and doesn't need to be duplicated here as well.

Reply:
Yeah I was thinking that too. I think this test was added back in the early days of fixed vector support before there was better rounded test coverage. I'll remove the fcmps in an NFC and rebase.